sql: syntax support of DROP FUNCTION and ALTER FUNCTION statements

chengxiong-ruan opened the pull request:
This pr contains 3 commits:
commit (1) adds syntax support for DROP FUNCTION statement.
commit (2) adds syntax support for ALTER FUNCTION options statement.
commit (3) adds syntax support for ALTER FUNCTION statements for rename, set owner, set schema and extension dependency (which doesn't really matter but still need to teach the parser a bit)

chengxiong-ruan committed:
sql: syntax support of DROP FUNCTION

This commit adds syntax support for DROP FUNCTION but leave
backend as unimplemented.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/85487)
<!-- Reviewable:end -->

chengxiong-ruan committed:
sql: syntax support for ALTER FUNCTION options

This commit adds syntax support for ALTER FUNCTION...action
statement which alters function options.

Release note: None

chengxiong-ruan committed:
sql: syntax support for other ALTER FUNCTION statements

This commit adds syntax support for 4 ALTER FUNCTION statements
including rename, owner, set schema and extension.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 18 of 18 files at r1, 18 of 18 files at r2, 18 of 18 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85487#-:-N8abS8n3hw4L0sXRPQQ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @chengxiong-ruan)

___
*[`-- commits` line 20 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/85487#-N8aaykp4t6tqhLq8oen):*
nit: `FUNCTIOn`


<!-- Sent from Reviewable.io -->

chengxiong-ruan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85487)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @chengxiong-ruan)

___
*[`-- commits` line 20 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/85487#-N8aaykp4t6tqhLq8oen):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

nit: `FUNCTIOn`
</blockquote></details>

fixed, thanks.


<!-- Sent from Reviewable.io -->

chengxiong-ruan commented:
TFTR!
bors r+
